### PR TITLE
[4.0] Meta Description character count

### PR DESCRIPTION
**Pull request by brianteeman**
redo of #25262

A request was made to have a character count when entering a meta description. As a Proof of Concept I have done this and applied it to the Site Meta Description in the global configuration as shown in the screenshot. (If accepted it is quick to add it to all the meta description fields)

image

This can be applied to any textarea field by adding `charcounter="true"` to the field definition in the xml.

The character count has a default maximum of 200 characters but this can be configured in the field xml by adding `maxlength="160"` (this is the setting for this POC)

Accessibility is obviously very important and this PR addresses this as well. The meta description text is announced when the user focuses on the input together with the string to indicate the characters used. Obviously a live count announced by the screenreader would be distracting so it waits 2000 milliseconds before updating the user with the screen reader user with the current count.

The string `86 characters remaining of 160 characters.` is a regular joomla language string except the numbers are replaced  by the script. In this example I chose
`"{remaining} characters remaining of {maxlength} characters."`. Available placeholders are {remaining}, {maxlength}, {length}

### Documentation
Needs updating to document the new charcounter option

### CODE REVIEW

File: build/build-modules-js/init.es6.js [hunk 1]
```diff
@@ -259,6 +259,16 @@ const copyFiles = (options) => {
       Fs.writeFileSync(chosenPath, ChosenJs, { encoding: 'UTF-8' });
     }
 
+    // Append initialising code to the end of the Short-and-Sweet javascript
+    if (packageName === 'short-and-sweet') {
+      const dest = Path.join(mediaVendorPath, vendorName);
+      const shortandsweetPath = `${dest}/${options.settings.vendors[packageName].js['dist/short-and-sweet.min.js']}`;
+      let ShortandsweetJs = Fs.readFileSync(shortandsweetPath, { encoding: 'UTF-8' });
+      ShortandsweetJs = ShortandsweetJs.concat(`document.addEventListener('DOMContentLoaded', function()`
```
[the remark this answers is not in the document]
Comment (C-Lodder) [reply build/build-modules-js/init.es6.js]: Why are you concatenating the event listener with the dependency code?

[the remark this answers is not in the document]
Comment (brianteeman) [reply build/build-modules-js/init.es6.js]: TBH I dont remember the reason. I did the original PR ages ago.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 5eef87d0eb31705b82e059c53c888a223bb9c9ff<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27359">issues.joomla.org/tracker/joomla-cms/27359</a>.</sub>

**Comment by richard67** [comment]
@brianteeman Is there a way to fix the hound?

**Comment by brianteeman** [comment]
hound is happy now

**Comment by richard67** [comment]
Now the drone is mad .. but I think that's not related to your PR.

**Comment by brianteeman** [comment]
lets see if drone is any happier now

**Comment by rdeutz** [comment]
Why this PR has changed 900+ files and over 200 commits, looks not right to me

**Comment by brianteeman** [comment]
something went wrong when I updated the branch :(

**Comment by brianteeman** [comment]
closing this while I fix the pr

**Comment by brianteeman** [comment]
Had to force push to fix the problem which prevents being able to re-open

See #27804

**Pull request closed without merging**